File: Python/1448. Count Good Nodes in Binary Tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        ans = 0
        qu = [[root,""]]
        while len(qu) > 0:
            pa = qu.pop(0)
            node = pa[0]
            psf = pa[1]
            psf += " "+str(node.val)
            if self.isGood(psf,node.val):
                ans+=1
                
            if(node.left):
                qu.append([node.left,psf])
            if node.right:
                qu.append([node.right,psf])
                
        return ans
    
    def isGood(self,s,X):
        s = s.split(" ")
        i = 0
        while i < len(s) and s[i] == "":
            i+=1
            
        while i < len(s):
            if int(s[i]) <= X:
                i+=1
                continue
            return False
        
        return True

File: Java/1448. Count Good Nodes in Binary Tree.java
class Pair{
    TreeNode node;
    String psf;
    Pair(TreeNode n,String p){
        node  = n;
        psf = p;
    }
}
class Solution {
    public int goodNodes(TreeNode root) {
        Queue<Pair> qu = new ArrayDeque<>();
        qu.add(new Pair(root,""));
        int ans = 0;
        while(qu.size()>  0){
            Pair p = qu.remove();
            TreeNode node = p.node;
            String psf = p.psf;
            psf += " "+Integer.toString(node.val);
            if(isGood(psf,node.val))
                ans++;
            if(node.left!=null)
                qu.add(new Pair(node.left,psf));
            if(node.right!=null)
                qu.add(new Pair(node.right,psf));
        }
        return ans;
    }
    private boolean isGood(String s,int X){
        String[] a = s.split(" ");
        int i=1;
        while(i<a.length){
            if(Integer.parseInt(a[i])<=X){
                i++;
                continue;
            }
            return false;
        }
        return true;
    }
}
